src/validate_i18n.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import re
+
+from flask import request
+from werkzeug.exceptions import abort
+from functools import wraps
+
+DEFAULT_YEAR = '2019'
+DEFAULT_LANG = 'en'
+SUPPORTED_YEARS = {
+    '2019': ('en', 'ja')
+}
+
+
+def validate_i18n(func):
+    @wraps(func)
+    def decorated_function(*args, **kwargs):
+        lang = kwargs.get('lang')
+        year = kwargs.get('year')
+
+        updated_kwargs = validate(lang, year)
+
+        kwargs.update(updated_kwargs)
+
+        return func(*args, **kwargs)
+
+    return decorated_function
+
+
+def validate(lang, year):
+    if year is None:
+        logging.debug('Defaulting the year to: %s' % year)
```

Comment:
When this is running on App Engine, do you know if these logs are available in Stackdriver or similar?

Reply:
I am not 100% certain, but the docs lead me to believe so: https://cloud.google.com/logging/docs/setup/python